airflow/serialization/serialized_objects.py

```diff
@@ -187,12 +190,12 @@ def _serialize(cls, var: Any) -> Any:  # Unfortunately there is no support for r
                     {str(k): cls._serialize(v) for k, v in var.items()},
                     type_=DAT.DICT
                 )
-            elif isinstance(var, k8s.V1Pod):
+            elif HAS_KUBERNETES and isinstance(var, k8s.V1Pod):
                 json_pod = PodGenerator.serialize_pod(var)
                 return cls._encode(json_pod, type_=DAT.POD)
             elif isinstance(var, list):
                 return [cls._serialize(v) for v in var]
-            elif isinstance(var, k8s.V1Pod):
+            elif HAS_KUBERNETES and isinstance(var, k8s.V1Pod):
```

Comment:
This block is redundant actually; no need to test for k8s.V1Pod _twice_.

Reply:
PR: https://github.com/apache/airflow/pull/11602